distribution/xfer/download.go

```diff
@@ -33,10 +43,16 @@ func (ldm *LayerDownloadManager) SetConcurrency(concurrency int) {
 
 // NewLayerDownloadManager returns a new LayerDownloadManager.
 func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int) *LayerDownloadManager {
-	return &LayerDownloadManager{
-		layerStore: layerStore,
-		tm:         NewTransferManager(concurrencyLimit),
+	ldm := &LayerDownloadManager{
+		layerStore:               layerStore,
+		tm:                       NewTransferManager(concurrencyLimit, filepath.Join(os.Getenv("TMPDIR"), "resumable_layer_downloads")),
+		cachedLayerRefs:          make(map[string]layer.Layer),
+		imagePullCheckpointsPath: filepath.Join(os.Getenv("TMPDIR"), "resumable_image_downloads"),
```

Comment:
I meant that I wanted the daemon to be able to resume a download if it's killed in the middle of a download and started again. If we use a random temporary directory, we don't have a way to know which temporary directory was used for which download after we restart.